api/oc_ri.c

```diff
@@ -1697,7 +1697,7 @@ oc_ri_free_client_cbs_by_endpoint(const oc_endpoint_t *endpoint)
     if (!cb->multicast && !cb->discovery && cb->ref_count == 0 &&
         oc_endpoint_compare(&cb->endpoint, endpoint) == 0) {
       cb->ref_count = 1;
-      notify_client_cb_timeout(cb);
+      notify_client_cb_with_code(cb, OC_CONNECTION_CLOSED);
```

Comment:
Both `oc_ri_free_client_cbs_by_mid` and `oc_ri_free_client_cbs_by_endpoint` are public API, so they can be called whenever, not just when connection is closed. So I think `OC_CONNECTION_CLOSED` doesn't always correctly describe what's happening. I guess both functions cancel the callback, so maybe something like `OC_CANCELLED` ? 
Or maybe we should add a status code parameter, so they caller can determine which code should be received in the cancelled callback. Although that would be an API change.

Reply:
fixed